pkgs/development/compilers/temurin-bin/jdk-darwin-base.nix

```diff
@@ -40,16 +40,19 @@ let
     installPhase = ''
       cd ..
 
-      mv $sourceRoot $out
+      mkdir -p $out/Library/Java/JavaVirtualMachines
+
+      bundle=$out/Library/Java/JavaVirtualMachines/${name-prefix}-${lib.versions.major version}.jdk
+      mv $sourceRoot $bundle
 
       # jni.h expects jni_md.h to be in the header search path.
-      ln -s $out/Contents/Home/include/darwin/*_md.h $out/Contents/Home/include/
+      ln -s $bundle/Contents/Home/include/darwin/*_md.h $bundle/Contents/Home/include/
 
       # Remove some broken manpages.
       # Only for 11 and earlier.
-      [ -e "$out/Contents/Home/man/ja" ] && rm -r $out/Contents/Home/man/ja
+      [ -e "$bundle/Contents/Home/man/ja" ] && rm -r $bundle/Contents/Home/man/ja
 
-      ln -s $out/Contents/Home/* $out/
+      ln -s $bundle/Contents/Home/* $out/
```

Comment:
I can't see anything wrong with any of this. I just don't like it for "I don't like Darwin" reasons. Wacky paths.